Mathlib/NumberTheory/Padics/Harmonic.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2023 Koundinya Vajjha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Koundinya Vajjha
+-/
+
+import Mathlib.NumberTheory.Padics.PadicIntegers
+import Mathlib.Data.Int.Log
+/-!
+
+The nth Harmonic number is not an integer. We formalize the proof using
+2-adic valuations. This proof is due to Kürschák.
+
+Reference:
+https://kconrad.math.uconn.edu/blurbs/gradnumthy/padicharmonicsum.pdf
+
+-/
+
+open BigOperators
+
+/-- The nth-harmonic number defined as a finset sum of consecutive reciprocals. -/
+def harmonic : ℕ → ℚ := fun n => ∑ i in Finset.range n, 1 / (i + 1)
+
+lemma harmonic_pos (n : ℕ) (Hn : n ≠ 0) : 0 < harmonic n :=
```

Comment:
```suggestion
lemma harmonic_pos {n : ℕ} (Hn : n ≠ 0) : 0 < harmonic n :=
```
an argument should normally be implicit when followed by another argument that pins it down. in this case this probably means one of the uses below will break but that should be easily fixable

Reply:
thanks! missed that one